mne/externals/pymatreader/pymatreader.py

```diff
@@ -117,7 +126,9 @@ def _hdf5todict(hdf5_object, variable_names=None, ignore_fields=None):
         data = hdf5_object.value
         if isinstance(data, numpy.ndarray) and data.dtype == numpy.dtype('object'):
             data = [hdf5_object.file[cur_data] for cur_data in data.flatten()]
-            data = numpy.squeeze(_hdf5todict(data)).T
+            data = _hdf5todict(data)
+            if isinstance(data, numpy.ndarray):
+                data = numpy.squeeze(data).T
```

Comment:
Usually `squeeze` is not a great idea -- if someone stores an array as `(1, 5)` because it happens to have a single element this becomes `(5,)` for example. And the problem gets worse for higher dimensional data. Why do you need to `squeeze`?

Reply:
... though I guess you `squeeze` below, too, so I guess this is already being done and is hopefully okay! I'll merge if everything is good here and give it a try in my PR.

Reply:
we had problems with extra dimensions being produced by h5py, so squeeze was the obvious thing to do and it has worked so far for all our use-cases.

and i also did not introduce this behavior in the latest version. it has been there ever since.

however, i am of course open to other suggestions. but if i recall correctly, we did not see any way to tell apart deliberate extra dimensions from those created by h5py.